File: Factory/PizzaStore.CS/PizzaStore/Store.cs
namespace PizzaStore;


public abstract class PizzaStore
{
    public Pizza OrderPizza(String type)
    {
        Pizza pizza;
        pizza = this.CreatePizza(type);
        pizza.Prepare();
        pizza.Bake();
        pizza.Cut();
        pizza.Box();
        return pizza;
    }

    protected abstract Pizza CreatePizza(String type);
}

public class NYPizzaStore : PizzaStore
{
    protected override Pizza CreatePizza(String type)
    {
        Pizza pizza = null;
        PizzaIngredientFactory ingredientFactory = new NYPizzaIngredientFactory();

        if (type == "cheese")
        {
            pizza = new CheesePizza(ingredientFactory);
            pizza.Name = "New York Style Cheese Pizza";
        }
        // else if (type == "veggie")
        //     return new NYStyleVeggiePizza();
        // else if (type == "clam")
        //     return new NYStyleClamPizza();
        // else if (type == "pepperoni")
        //     return new NYStylePepperoniPizza();
        return pizza;
    }
}

// public class ChicagoPizzaStore : PizzaStore
// {
//     protected override Pizza CreatePizza(String type)
//     {
//         if (type == "cheese")
//             return new ChicagoStyleCheesePizza();
//         // else if (type == "veggie")
//         //     return new NYStyleVeggiePizza();
//         // else if (type == "clam")
//         //     return new NYStyleClamPizza();
//         // else if (type == "pepperoni")
//         //     return new NYStylePepperoniPizza();
//         else return null;
//     }
// }

File: Factory/Python/src/Store.py
from abc import ABC, abstractmethod

from . import Pizza
from . import Ingredients


class PizzaStore(ABC):
    def order_pizza(self, pizza_type: str) -> Pizza.Pizza:
        pizza: Pizza.Pizza = self.create_pizza(pizza_type)
        pizza.prepare()
        pizza.bake()
        pizza.cut()
        pizza.box()
        return pizza

    @abstractmethod
    def create_pizza(self, pizza_type: str) -> Pizza.Pizza:...


class NYPizzaStore(PizzaStore):
    def create_pizza(self, pizza_type: str) -> Pizza.Pizza:
        pizza: Pizza.Pizza = None
        ingredient_factory = Ingredients.NYPizzaIngredientFactory()

        if pizza_type == "cheese":
            pizza = Pizza.CheesePizza(ingredient_factory)
            pizza.name = "New York Style Cheese Pizza"
        # elif (type == "veggie")
            # return new NYStyleVeggiePizza()
        # elif (type == "clam")
            # return new NYStyleClamPizza()
        # elif (type == "pepperoni")
            # return new NYStylePepperoniPizza()
        return pizza